### PR TITLE
Undefined initialValue - Field dont update if its not yet registered

### PR DESCRIPTION
**Pull request by Vidlec**

### CODE REVIEW

**Comment by Vidlec** [comment]
Addresses #134 - Undefined initialValue

File: stories/DefaultForm.jsx [hunk 1]
```diff
@@ -200,12 +200,19 @@ export default class DefaultForm extends Component {
             </div>
 
             <MyInput
-              name="foo"
+              name="initialValue"
               initialValue="Pooper"
               placeholder="My placeholder"
               disabled={ isDisabled }
               required />
```
Comment (kettanaito) [inline stories/DefaultForm.jsx]: Please add the test instead of a story.

You may add it under Field.spec.js, if any. Thanks.

File: src/Form.jsx [hunk 2]
```diff
@@ -161,6 +161,10 @@ export default class Form extends React.Component {
   updateField = ({ fieldPath, fieldProps: directFieldProps, propsPatch = null }) => {
     const { fields } = this.state;
     const fieldProps = directFieldProps || fields.getIn([fieldPath]);
+
+    /* Do not update field if its not registered yet */
+    if (!fieldProps) return;
```
Comment (kettanaito) [inline src/Form.jsx]: Could you please describe me why does this happen?

When updateField is called in this bug scenario? Which lifecycle event triggers that? How is that related to initialValue?

How come field is not yet registered?

**Comment by Vidlec** [comment]
Okey so this is what I think is happening:
`Field.registerWith` is called, which in turn calls `Form.registerField`. 

Field is then added to state. And Field receives props.

But for some reason all the fields fire `willRecieveProp`s, even more strangely with context props, which (If I understand correctly) they shouldn't have since they are not yet registered.

Anyway, `willRecieveProps` will trigger `this.context.updateField`. 
In `this.context.updateField` we are taking the fields from the state, but they are not there yet!
Only the first one which triggered `willRecieveProps` on all fields...

**Comment by kettanaito** [comment]
Thanks for the investigation!

Please let me try one thing to test this as well before merging your fix. There may be a chance that the issue is elsewhere. Sorry for inconvenience.

File: src/Form.jsx (hunk 2 above)
Comment (kettanaito) [reply src/Form.jsx]: Also please take a look at each event handler on Form level - handleFieldChange, etc. 

All event handlers are bypassed before the field is actually registered. So this issue is somewhat unclear to me. I would suggest more investigation.

I want to look into this once I have some spare time.

**Comment by kettanaito** [comment]
Okay, after I've setup another field with undefined initialValue, as you suggested @Vidlec, I am able to reproduce the issue.

## The reason
The issue is happening due to the `utils.getPropsPatch()` method. When `initialValue: undefined`, it returns it as an update patch, since it may happen that the prop has been changed to have `undefined` value. This change should be reflected in the internal `fields` map as well. That is what triggers `updateField()`.

## The solution
Solution would be to set proper initialValue on field registration, or proper comparison logic during `utils.getPropsPatch()`.

The solution you suggest doesn't solve the issue, may I close this pull request? Thank you.

**Comment by Vidlec** [comment]
Yes, closing
